model-optimizer/mo/utils/versions_checker.py

```diff
@@ -44,34 +44,74 @@ def check_python_version():
         return 1
 
 
-def parse_versions_list(required_fw_versions, version_list):
+def parse_versions_list(required_fw_versions, version_list, env_params):
     """
     Please do not add parameter type annotations (param:type).
     Because we import this file while checking Python version.
     Python 2.x will fail with no clear message on type annotations.
 
-    Parsing requirements versions
+    Parsing requirements versions for a dependency,
+    if environment version (python_version, etc.) is satisfied
     :param required_fw_versions: String with fw versions from requirements file
     :param version_list: List for append
+    :param env_params: a dictionary of environment variables
     :return: list of tuples of strings like (name_of_module, sign, version)
 
+    Examples of required_fw_versions:
+    'tensorflow>=1.15.2,<2.0; python_version < "3.8"'
+    'tensorflow>=2.0'
+
     Returned object is:
     [('tensorflow', '>=', '1.2.0'), ('networkx', '==', '2.1'), ('numpy', None, None)]
     """
 
     line = required_fw_versions.strip('\n')
     line = line.strip(' ')
     if line == '':
-        return []
-    splited_versions_by_conditions = re.split(r"==|>=|<=|>|<", line)
+        return version_list
+    splited_requirement = line.split(";")
+
+    # check environment marker
+    if len(splited_requirement) > 1:
+        env_req = splited_requirement[1]
+        splited_env_req = re.split(r"==|>=|<=|>|<", env_req)
+        splited_env_req = [l.strip(',') for l in splited_env_req]
+        env_marker = splited_env_req[0].strip(' ')
+        if env_marker == 'python_version' and env_marker in env_params:
+            installed_python_version = env_params['python_version']
+            env_req_version_list = []
+            splited_required_versions = re.split(r",", env_req)
+            for i, l in enumerate(splited_required_versions):
+                comparisons = ['==', '>=', '<=', '<', '>']
+                for comparison in comparisons:
+                    if comparison in l:
+                        required_version = splited_env_req[i + 1].strip(' ').replace('"', '')
+                        env_req_version_list.append((env_marker, comparison, required_version))
+                        break
+            exit_code = 0
+            not_satisfied_list = []
+            for name, key, required_version in env_req_version_list:
+                exit_code = version_check(name, installed_python_version, required_version,
+                                          key, not_satisfied_list, exit_code)
```

Comment:
the `exit_code` should be removed as a parameter to the `version_check` function since it is not needed to be a parameter. And since the `exit_code` is not used as a result then let's remove it.

Reply:
Fixed.